src/harness/reference_models/ppa/ppa.py

```diff
@@ -0,0 +1,164 @@
+from reference_models.propagation import wf_hybrid
+from shapely import geometry, ops
+
+THRESHOLD_PER_10MHZ = -96
+RX_HEIGHT = 1.5
+MAX_ALLOWABLE_EIRP_PER_10_MHZ_CAT_A = 30.
+MAX_ALLOWABLE_EIRP_PER_10_MHZ_CAT_B = 47.
+census_tract_driver = census_tract.CensusTractDriver()
+nlcd_driver = nlcd.NlcdDriver()
+
+
+def _CalculateDbLossForEachPointAndGetContour(install_param, eirp_capability, antenna_gain,
+                                              cbsd_region_type, latitudes, longitudes):
+  db_loss = np.zeros(len(latitudes), dtype=np.float64)
+  for index, lat_lon in enumerate(zip(latitudes, longitudes)):
+    lat, lon = lat_lon
+    db_loss[index] = wf_hybrid.CalcHybridPropagationLoss(install_param['latitude'],
+                                                         install_param['longitude'],
+                                                         install_param['height'],
+                                                         lat, lon, RX_HEIGHT,
+                                                         install_param['indoorDeployment'],
+                                                         reliability=0.5,
+                                                         region=cbsd_region_type).db_loss
+  index_cond, = np.where(eirp_capability + antenna_gain - db_loss > THRESHOLD_PER_10MHZ)
+  return index_cond.shape[0] * 0.2
+
+
+def _HammingFilter(x, window_len=15):
+  """Process a full radial at a given azimuth for all distance ranges"""
+  s = np.r_[x[-(window_len / 2):], x, x[0:window_len / 2]]
+  w = np.hamming(window_len)
+  y = np.convolve(w / w.sum(), s, mode='valid')
+  return y
+
+
+def _GetPolygon(device):
+  """Returns the PPA contour for a single CBSD device, as a shapely polygon."""
+  install_param = device['installationParam']
+  eirp_capability = install_param.get('eirpCapability',
+                                      MAX_ALLOWABLE_EIRP_PER_10_MHZ_CAT_A
+                                      if device['cbsdCategory'] == 'A' else
+                                      MAX_ALLOWABLE_EIRP_PER_10_MHZ_CAT_B)
+
+  # Compute all the Points in 0-359 every 200m upto 40km
+  distances = np.arange(0.2, 40.1, 0.2)
+  azimuths = np.arange(0.0, 360.0)
+  latitudes, longitudes, _ = zip(*[vincenty.GeodesicPoints(install_param['latitude'],
+                                                           install_param['longitude'],
+                                                           distances, azimuth)
+                                   for azimuth in azimuths])
+  # Compute the Gain for all Direction
+  antenna_gains = antenna.GetStandardAntennaGains(azimuths,
+                                                  install_param['antennaAzimuth']
+                                                  if 'antennaAzimuth' in install_param.keys() else None,
+                                                  install_param['antennaBeamwidth'],
+                                                  install_param['antennaGain'])
```

Comment:
The comment from Ariful Hannan has not been taken into account. Because we compute the signal level as:
 ``` eirp + antenna_gain - pathloss ```
so we are counting twice the global antenna gain.
Simple solution is to change this line as:
```
norm_antenna_gains = antenna_gains = antenna.GetStandardAntennaGains(azimuths,
                                                  install_param['antennaAzimuth']
                                                  if 'antennaAzimuth' in install_param.keys() else None,                                                  
                                                 install_param['antennaBeamwidth'],
                                                 0)